x-pack/plugins/fleet/server/services/epm/elasticsearch/transform/install.ts

```diff
@@ -383,37 +401,43 @@ const installTransformsAssets = async (
       transforms,
       destinationIndexTemplates,
       transformsSpecifications,
-      aliasesRefs,
       transformsToRemove,
       transformsToRemoveWithDestIndex,
     } = processTransformAssetsPerModule(
       installablePackage,
       installNameSuffix,
       transformPaths,
-      previousInstalledTransformEsAssets
+      previousInstalledTransformEsAssets,
+      username
     );
 
-    // ensure the .latest alias points to only the latest
-    // by removing any associate of old destination indices
-    await Promise.all(
-      aliasesRefs
-        .filter((a) => a.endsWith(TRANSFORM_DEST_IDX_ALIAS_LATEST_SFX))
-        .map((alias) => deleteAliasFromIndices({ esClient, logger, alias }))
-    );
+    // By default, for internal Elastic packages that touch system indices, we want to run as internal user
+    // so we set runAsKibanaSystem: true by default (e.g. when run_as_kibana_system set to true/not defined in yml file).
+    // If package should be installed as the logged in user, set run_as_kibana_system: false,
+    // and pass es-secondary-authorization in header when creating the transforms.
+    const secondaryAuth = await generateTransformSecondaryAuthHeaders({
```

Comment:
It's inefficient to generate these credentials if they aren't going to be used lower down the function.

Reply:
As discussed, will address this in the follow-up.